docs/wallets.md

```diff
@@ -0,0 +1,57 @@
+# For Wallet Developers
+
+codex32 is a new format for BIP32 master seeds. It is essentially a replacement for
+BIP39 or SLIP39 seed words, and the user workflow should be much the same, except
+that:
+
+* The character set is different (bech32 characters rather than a wordlist).
+* Seeds may be encoded across multiple shares, rather than a single string.
+* It is possible to do specific error detection/correction during entry.
+
+There are two levels of wallet support:
+
+* The ability to import seeds/shares; here we essentially just have recommendations about dealing with errors.
+* The ability to generate seeds/shares on the device; here our guidance is more involved.
+
+We encourage every wallet to support importing seeds and shares, since
+* the technial difficulty is low (roughly on par with that of supporting Segwit addresses, plus optional error-correction support);
+* the added functionality is isolated from the rest of the wallet (once the seed is imported you don't care where it came from); and
+* beyond correctness of the code, there is little risk (no need to source randomness or execute potentially variable-time algoritms).
+
+Supporting seed generation is a little more involved so the tradeoff between
+implementation complexity and user value is less clear, especially since the
+Codex provides users instructions on doing generation themselves.
+
+## Import Support
+
+codex32 shares may be any length between 128 and 512 bits.
+Wallets should support import of 128- or 256-bit seeds; other lengths are optional.
+
+128-bit seeds are 48 characters in length, including the `MS1` prefix.
+256-bit seeds are 74. For other bitlengths, see the BIP.
+
+The process for entering shares is:
+
+1. The user should select the bit length of the import before entering any actual data.
```

Comment:
Disagree.  User's don't necessarily know anything about bit lengths, or what they are.  It doesn't appear in the shares themselves.

Reply:
Agreed. By recommending electronic generations use 16, 32 or 64 byte `master_seeds` we can guess the correct length. And even for odd-ball hand generated 17-byte sets, the checksum will almost always confirm it was not a typo. User should have to confirm an "unusual" length is correct (matches what is written) however after entering it. Since the checksum can't give absolute guarantees for insertion errors. After the 1st share is validated, then similar to the header, the new length can be used for error correction of subsequent shares.